pkg/pool-utils/contracts/external-fees/ProtocolFeeCache.sol

```diff
@@ -84,11 +92,11 @@ abstract contract ProtocolFeeCache is RecoveryMode {
         }
 
         if (feeType == ProtocolFeeType.SWAP) {
-            return _cache.swapFee;
+            return _feeCache.decodeUint(_SWAP_FEE_OFFSET, _FEE_TYPE_CACHE_WIDTH);
         } else if (feeType == ProtocolFeeType.YIELD) {
-            return _cache.yieldFee;
+            return _feeCache.decodeUint(_YIELD_FEE_OFFSET, _FEE_TYPE_CACHE_WIDTH);
         } else if (feeType == ProtocolFeeType.AUM) {
-            return _cache.aumFee;
+            return _feeCache.decodeUint(_AUM_FEE_OFFSET, _FEE_TYPE_CACHE_WIDTH);
         } else {
```

Comment:
If the if-else chain looked for the offset and you then had a single `decode` call, that'd probably save more bytecode.

Reply:
Done.